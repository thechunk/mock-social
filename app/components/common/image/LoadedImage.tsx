import * as React from 'react';
import {PureComponent} from "react";
import {ActivityIndicator, Animated, Image, ImageProps} from "react-native";
import styles from './styles';
import {Loader} from "../loader/index";

export default class LoadedImage extends PureComponent<ImageProps & ILoadedImage, ILoadedImageState> {
    constructor(props: ImageProps) {
        super(props);
        this.state = {opacity: new Animated.Value(0), loading: true};
        this.onImageLoaded = this.onImageLoaded.bind(this);
    }

    onImageLoaded() {
        Animated.timing(this.state.opacity, {
            toValue: 1,
            duration: 200
        }).start(() => this.setState(state => ({...state, loading: false})));
    }

    render() {
        return (
            <>
                {this.state.loading && this.props.showSpinner
                    ? <Loader loading={true} style={styles.imageLoader} />
                    : null}
                <Animated.View style={[this.props.style, {opacity: this.state.opacity}]}>
                    <Image {...this.props as ImageProps} onLoadEnd={this.onImageLoaded}/>
                </Animated.View>
            </>
        );
    }
}