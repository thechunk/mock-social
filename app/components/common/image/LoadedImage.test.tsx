import * as React from 'react';
import {shallow} from "enzyme";
import LoadedImage from "./LoadedImage";

describe('LoadedImage', () => {
    it('renders correctly', () => {
        expect(shallow(<LoadedImage source={{uri: 'uri'}}/>)).toMatchSnapshot();
    });
});