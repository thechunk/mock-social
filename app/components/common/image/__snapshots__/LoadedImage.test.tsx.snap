// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadedImage renders correctly 1`] = `
<Fragment>
  <AnimatedComponent
    style={
      Array [
        undefined,
        Object {
          "opacity": 0,
        },
      ]
    }
  >
    <Image
      onLoadEnd={[Function]}
      source={
        Object {
          "uri": "uri",
        }
      }
    />
  </AnimatedComponent>
</Fragment>
`;
